### ISSUE TITLE
questions about the weights of different losses

### ISSUE THREAD

**Issue by icoz69** [CQFIO/PhotographicImageSynthesis#12]
in the demo 256 code, the weights of different losses are 1,1/1.6,1/2.3,1/2.8,10/0.5. where do these hyperparameters come from? 
in the paper, it says they are "inverse of the number of elements in each layer', what do you mean by "number of elements", and how to calculate the weights above? 
looking forward to ur reply, thank you

**Comment by CQFIO** [comment]
If you just the weights listed there, it should be fine.

I use tf.reduce_mean to compute the average. The weights 1,1/1.6,1/2.3,1/2.8,10/0.5 are computed from the average l0,...,l5 in the 100 epoch.

**Comment by MORIZHIJIANDX** [comment]
Sorry, I didn't get it. I have some questions.
1.At the beginning of the training, the loss weights are "initialized to the inverse of the number of elements in each layer", "the number of elements in each layer" refers to the number of elements of the future map mi,which is extracted by the layer i ?(or the elements of weight of layer i ?)
So the initial value of the loss weight wi is:   wi = 1/count(mi)
2.Keep the loss weight constant and then train 100 times.In the last training we got the loss L(ik) of layer i and image k.
So the final weight wi is: wi = ( Li1 + Li2 +... + Lin)/n 
I am not sure whether I understand it correctly.Look forward to your reply.

**Comment by icoz69** [comment]
@CQFIO 
thanks for your reply.
what do  l0,...,l5 mean?
and how could the weights be learnable? losses are always positive, so weighting will keep decreasing during the training.
